solarforecastarbiter/validation/validator.py

```diff
@@ -557,22 +557,61 @@ def check_poa_clearsky(poa_global, poa_clearsky, kt_max=1.1):
 
 
 @mask_flags('NIGHTTIME')
-def check_irradiance_day_night(solar_zenith, max_zenith=87):
+def check_irradiance_day_night(
```

Comment:
@cwhanse if you put this into pvanalytics, would you want a single function like the one shown here, or would you want two functions: `check_irradiance_day_night` (the old simple function) and `check_irradiance_day_night_interval` (just the else clause)? 

Now that I've implemented it, the single function feels unwieldy.

Reply:
I think a single function that 1) upscales the input Series index to 1 minute 2) applies the zenith angle criteria to  determine day/night for each minute and 3) reports day/night for each interval in the input Series using some kwarg-controlled rule. With that design the zenith angle criteria and the day/night vote could be controllable. I'm guessing that some helper functions will fall out that could be re-used elsewhere, for up/downscaling.

It would be a nice complement to the day/night labeler from PVFleets, which doesn't ask for location, in case that is not yet trusted.

@wfvining?

Reply:
Are you describing the higher level function [`tasks._solpos_night_resample`](https://github.com/SolarArbiter/solarforecastarbiter-core/pull/579/files#diff-76a80c5cfa7f29028e290445c9479381R48), minus the Arbiter specific things (the datamodel, the bitmask instead of the simple bool series, and the returned solar position)?

The PVFleets function doesn't work here - we need a function that works on a day or less of data.

Reply:
I think two different functions is a little better. I spent some time trying to convince myself that a single function that defaults to the original simplistic behavior is better, but there are too many opportunities for users to get unexpected behavior.

Scrolling down I see @wholmgren put it much more concisely "a function should do one thing"

Reply:
Yes, I was describing what I see could be a general use function for labelling day/night using solar zenith, that accounts for time interval. I don't see much value in having two functions which a user must choose based on the time interval of the data. IMO it would be better to offer a single function that handles the time interval.

I was only pointing out that the PVFleets function has a narrow set of use cases and other day/night labelers would be good additions to pvanalytics.

Reply:
Thanks for the input @wfvining and @cwhanse. It helped me make the code more a bit more generic and hopefully easier to adapt to other situations. I'm going to forge ahead with the current approach in the arbiter, but happy to help (or stay out of it) if/when you take it up in pvanalytics.